packages/frontend/src/pages/campaigns/filters/checkboxes-filter/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { Template } from "@carrot-kpi/sdk";
+import { Checkbox, Typography } from "@carrot-kpi/ui";
+import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
+import { FiltersLoading } from "../loading";
+import {
+    CheckboxesObj,
+    generateGroupId,
+    getDefaultCheckboxes,
+    getDefaultSelected,
+} from "../utils";
+
+interface CheckboxesFilterProps {
+    items: Template[];
+    groupId: string;
+    title: string;
+    loading: boolean;
+    selected: Set<number>;
+    setSelected: (newSelected: Set<number>) => void;
+    defaultChecked: (item: Template) => boolean;
+}
+
+export const CheckboxesFilter = ({
+    items,
+    title,
+    loading,
+    selected,
+    setSelected,
+    defaultChecked,
+    groupId,
+}: CheckboxesFilterProps) => {
+    const [checkboxes, setCheckboxes] = useState<CheckboxesObj>({});
+
+    const initializeCheckboxes = useCallback(() => {
+        if (items.length === 0) return;
+        if (Object.keys(checkboxes).length !== 0) return;
+
+        const defaultCheckboxes = getDefaultCheckboxes(items, defaultChecked);
+        setCheckboxes(defaultCheckboxes);
+        setSelected(getDefaultSelected(defaultCheckboxes));
+    }, [checkboxes, defaultChecked, items, setSelected]);
+
+    const getHtmlDataId = (element: HTMLInputElement) => {
+        const id = element.dataset.id;
+
+        if (id !== undefined) {
+            const parsedId = parseInt(id);
+            if (!isNaN(parsedId)) {
+                return parsedId;
+            }
+        }
+        return 0;
+    };
+
+    const handleCheckedChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
This and the previous functions not using `useCallback` cause a rerender on the checkbox each single time this component is called. I also think we can directly write the `getHtmlDataId` logic here to avoid an extra dependency in this `useCallback`able function.

Reply:
I added the `getHtmlDataId` logic directly there, also deleted the utils since became almost empty.

Reply:
I'm not getting the `useCallback`. It's not suppose the handleCheckbox to run every time a checkbox changes state?